cms/templatetags/cms_tags.py

```diff
@@ -67,8 +69,15 @@ def _get_page_by_untyped_arg(page_lookup, request, site_id):
         raise TypeError('The page_lookup argument can be either a Dictionary, Integer, Page, or String.')
     site = Site.objects._get_site_by_id(site_id)
     try:
+        warning_message = _('The page_url tag unexpectedly found multiple pages matching the request: %(page_lookup)s') % {'page_lookup': page_lookup}
         if 'pk' in page_lookup:
-            page = Page.objects.select_related('node').get(**page_lookup)
+            try:
+                page = Page.objects.select_related('node').get(**page_lookup)
+            except Page.MultipleObjectsReturned:
```

Comment:
I do not think this is a possible scenario: If the primary key `pk` is given, then exactly one (due to the uniqueness constraint on primary keys) or no `Page` will be found. The latter scenario is already taken care of in lines 99ff.
Therefore, I think this try except block is not necessary.